ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -280,9 +280,24 @@ func (r *RayJobReconciler) setRayJobIdAndRayClusterNameIfNeed(ctx context.Contex
 			rayJob.Status.JobId = utils.GenerateRayJobId(rayJob.Name)
 		}
 	}
+
 	if rayJob.Status.RayClusterName == "" {
 		shouldUpdateStatus = true
-		rayJob.Status.RayClusterName = utils.GenerateRayClusterName(rayJob.Name)
+		// if the clusterSelector is not empty, default use this cluster name
+		// we assume the length of clusterSelector is one
+		if len(rayJob.Spec.ClusterSelector) != 0 {
+			var useValue string
+			for _, value := range rayJob.Spec.ClusterSelector {
+				useValue = value
```

Comment:
em. I think cluster selector could be a dictionary of key:value which used to label selector one cluster .
In general, Let's say we have labels. key1=value1, key2=value2, we use the combination to choose the right cluster. (result could be a list)

If we assume a specific case that we always give `ray.io/cluster-name: value`, then we can use the name to get the target cluster. 

2nd option is acceptable to me but we need to give the assumption that `ray.io/cluster-name: value` has to be given in this case, otherwise it won't work. 


I didn't see the validation here and user can put aribitrary label here. Let's add some TODO and next steps at least. You can need to check label key instead of ignore it `_`

Reply:
it is a good idea. I have changed to use `ray.io/cluster` as the fixed key to get the cluster name